### ISSUE TITLE
Extreme uneven genome size in phased assemblies

### ISSUE THREAD

**Issue by jkreinz** [chhylp123/hifiasm#425]
Hi there,

I'm running a Hi-C integrated phased assembly for three different samples. Two of them have turned out well (estimated genome size ~650Mb, phased assemblies between 625 and 675, chromosome scale after scaffolding). However the third is showing extreme uneveness in genome size among the phased assemblies (400 vs 800 Mb) regardless of how I parameterize the run. I've tried -s values from 55 all the way down to 20, and it doesn't change the size distribution of the phased haplotypes at all (in contrast to the previous samples I've assembled where changing s did shift the sizes among phased haplotypes). I've tried modifying the genome size with --hg-size. I've also tried a combination of these parameters with no luck. I'm using hifiasm 0.18.8-r525. I've attached the last command I've implemented - any advice would be greatly appreciated! 

` hifiasm -o ${prefix}_s40_hapsize.asm -t32 -s 40 --h1 ${pathtohic}/${prefix}_1.fq.gz --h2 ${pathtohic}/${prefix}_2.fq.gz ${hifipath}/${hifi_pre}.fastq.gz --hg-size 650m &`

**Comment by chhylp123** [comment]
Have you checked the hom coverage threshold? See: https://hifiasm.readthedocs.io/en/latest/faq.html#why-one-hi-c-integrated-assembly-is-larger-than-another-one, also https://hifiasm.readthedocs.io/en/latest/faq.html#for-hi-c-integrated-assembly-why-the-assembly-size-of-both-haplotypes-are-much-larger-than-the-estimated-genome-size, https://hifiasm.readthedocs.io/en/latest/faq.html#how-can-i-tweak-parameters-to-improve-hi-c-integrated-assembly.

**Comment by jkreinz** [comment]
yes, I've adjusted the -s parameter --s-base parameter (both all the way down to 0.05), and set the expected --hg-size without improvement.  From what I can tell based on the command line printed histogram of KMER frequencies (see attachment), the inferred peak_hom looks correct (usually ~52) but I've also tried decreasing it and increasing slightly with little effect. Do you have any other suggestions for reducing the size difference between haplotypes? It seems surprising that default settings would work so well for only 2/3 samples.

![Screenshot 2023-04-20 at 3 45 35 PM](https://user-images.githubusercontent.com/15037621/233509142-02979d7d-69fc-40af-b5a6-8f08a1c329ca.png)

**Comment by jkreinz** [comment]
Seems like one of the other samples that worked was considerably more heterozygous (KMER hist attached below) but it still seems like there should be plenty of power to phase haplotypes? 
<img width="862" alt="Screenshot 2023-02-24 at 4 49 43 PM" src="https://user-images.githubusercontent.com/15037621/233511216-467d34e8-d363-4f73-a3d2-31e9fac5a50f.png">

**Comment by chhylp123** [comment]
Could you please show the assembly graph `p_utg.noseq.gfa`?

**Comment by jkreinz** [comment]
[Here is the p_utg.noseq.gfa file](https://drive.google.com/file/d/104mx0YxAmi1Lyr3Ckzh7D9gcTPcCEPcJ/view?usp=sharing) for a run with s=40 and hapsize = 650m that happened to give the total diploid genome size (summed across haplotype) closest to expected.

Thanks for your help!

**Comment by jkreinz** [comment]
Hi chhylp123, any insight into whether there is something strange about the assembly graph?

Thanks,
Julia

**Comment by chhylp123** [comment]
Sorry for the late reply. Can you rerun hifiasm with `-s 0.5` and '--hom-cov 52' with exisiting bin files? `-s 40` actually disable the homology detection. If it still does not work, could you please rerun hifiasm with `-s 0.5` and '--hom-cov 52' without HiC? This will give you two unphased haplotypes. If these two haplotypes are balanced, the HiC reads might have issues.
